apps/health_campaign_field_worker_app/integration_test/attendance_test.dart

```diff
@@ -0,0 +1,217 @@
+
+    await widgetTester.pumpAndSettle(const Duration(seconds: 1));
+
+    //check if we are on the manage attendance page
+    expect(
+      find.byType(ManageAttendancePage),
+      findsOneWidget,
+      reason: 'Manage Attendance Page not found',
+    );
+
+    expect(find.text('Open Register'), findsAtLeast(1));
+
+    await widgetTester.scrollUntilVisible(
+      find.text('Open Register').at(3),
+      1,
+    );
+    await widgetTester.pumpAndSettle(const Duration(seconds: 3));
+    await widgetTester.tap(find.text('Open Register').at(3));
+
+    await widgetTester.pumpAndSettle(
+      const Duration(seconds: 1),
+    ); //wait for the app to go to the next page
+
+    expect(
+      find.byType(AttendanceDateSessionSelectionPage),
+      findsOneWidget,
+    ); //check if we go to the next page
+    await widgetTester.pumpAndSettle(const Duration(seconds: 20));
+
+    // select Date
+    await widgetTester.tap(find.byType(DigitDateFormPicker));
+    await widgetTester.pumpAndSettle(const Duration(milliseconds: 500));
+    expect(find.byType(DigitDateFormPicker), findsAny);
+
+    await widgetTester.tap(find.text(testVariables['date']).last);
+    await widgetTester.pumpAndSettle(const Duration(milliseconds: 500));
+    await widgetTester.tap(find.text('Ok'));
+    await widgetTester.pumpAndSettle(const Duration(milliseconds: 500));
+
+    // select session
+    await widgetTester.tap(find.text(testVariables['session']));
+
+    await widgetTester.pumpAndSettle(const Duration(milliseconds: 500));
+    // // tap on the submit
+    await widgetTester.tap(find.text('Mark Attendance'));
+
+    await widgetTester.pumpAndSettle(const Duration(seconds: 2));
+
+    expect(
+      find.byType(MarkAttendancePage),
+      findsOneWidget,
+    ); //check if we go to the next page
+
+    //wait for attendees to load
+    await widgetTester.pumpAndSettle(const Duration(seconds: 2));
+
+    //mark attendance
+    //check if present word is displayed
+    await widgetTester.tap(find.byType(CircularButton).first);
+    await widgetTester.pumpAndSettle(const Duration(seconds: 2));
+    expect(find.text('Present'), findsOneWidget);
+
+    //check if absent word is displayed
+    await widgetTester.tap(find.byType(CircularButton).first);
+    await widgetTester.pumpAndSettle(const Duration(seconds: 2));
+    expect(find.text('Absent'), findsOneWidget);
+
+    //when the user taps on absent, the status must change back to present
+    await widgetTester.tap(find.byType(CircularButton).first);
+    await widgetTester.pumpAndSettle(const Duration(seconds: 2));
+    expect(find.text('Present'), findsOneWidget);
+    await widgetTester.pumpAndSettle(const Duration(seconds: 1));
+
+    //save and mark later option
+    await widgetTester.tap(find.text('Save and mark later'));
+    await widgetTester.pumpAndSettle(const Duration(seconds: 1));
+
+    //check if records are saved when we go back
```

Comment:
After this need to check if previous marked data is present, can't find that test case below